src/main/java/org/openrewrite/java/testing/junitassertj/JUnitFailToAssertJFail.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.junitassertj;
+
+import org.openrewrite.java.AutoFormat;
+import org.openrewrite.java.JavaIsoRefactorVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.RemoveUnusedImports;
+import org.openrewrite.java.tree.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.openrewrite.Formatting.EMPTY;
+import static org.openrewrite.Formatting.format;
+import static org.openrewrite.Tree.randomId;
+import static org.openrewrite.java.tree.MethodTypeBuilder.newMethodType;
+
+/**
+ * This is a refactoring visitor that will convert JUnit-style fail() to assertJ's fail().
+ *
+ * This visitor only supports the migration of the following JUnit 5 fail() methods:
+ *
+ * <PRE>
+ *  fail()                                  ->   fail("")
+ *  fail(String message)                    ->   fail(String message)
+ *  fail(String message, Throwable cause)   ->   fail(String message, Throwable cause)
+ *  fail(Throwable cause)                   ->   fail("", Throwable cause)
+ * </PRE>
+ *
+ * Note: There is an additional method signature in JUnit that accepts a StringSupplier as an argument. Attempts
+ *  to map this signature into assertJ's model obfuscates the original assertion.
+ */
+public class JUnitFailToAssertJFail extends JavaIsoRefactorVisitor {
+    private static final String JUNIT_QUALIFIED_ASSERTIONS_CLASS_NAME = "org.junit.jupiter.api.Assertions";
+    private static final String ASSERTJ_QUALIFIED_ASSERTIONS_CLASS_NAME = "org.assertj.core.api.Assertions";
+
+    /**
+     * This matcher finds the junit methods that will be migrated by this visitor.
+     */
+    private static final MethodMatcher JUNIT_FAIL_MATCHER = new MethodMatcher(
+            JUNIT_QUALIFIED_ASSERTIONS_CLASS_NAME + " fail(..)"
+    );
+
+    @Override
+    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method) {
+        J.MethodInvocation original = super.visitMethodInvocation(method);
```

Comment:
This may be a bit pedantic, but calling `super.visit...()` it's going to call all of the visit methods for the sub-AST elements, potentially returning a modified, non-original AST element if any sub-element has been altered.  _This_ visitor doesn't overload any other visit method, so you will always get the original back. But then, why call `super.visit...()` at all? Our AST elements are immutable so if you want to rename the parameter to original say 
```java
J.MethodInvocation original = method;
```
or rename the method argument.
```java
public J.MethodInvocation visitMethodInvocation(J.MethodInvocation original)
```